privacy-protections/storage-blocking/helpers/commonTests.js

```diff
@@ -1,7 +1,35 @@
 /* exported commonTests */
-/* global cookieStore */
+/* global cookieStore, THIRD_PARTY_TRACKER_ORIGIN, THIRD_PARTY_ORIGIN */
+
+function generateCookieHeaderTest (namePrefix, origin) {
+    return {
+        id: `${namePrefix} header cookie`,
+        store: (data) => {
+            return fetch(`${origin}/set-cookie?value=${data}`, { credentials: 'include' }).then(r => {
+                if (!r.ok) {
+                    throw new Error('Request failed.');
+                }
+            });
+        },
+        retrive: () => {
+            return fetch(`${origin}/reflect-headers`, { credentials: 'include' })
+                .then(r => r.json())
+                .then(data => data.headers.cookie.match(/headerdata=([0-9]+)/)[1]);
```

Comment:
One more thing, all of those tests are setting a `headerdata` cookie on the same set of two domains (THIRD_PARTY_TRACKER_ORIGIN, THIRD_PARTY_ORIGIN). Some of them will be set in a first party context, some of them in a third party context. So it may happen that a cookie will be set in a first-party context but be reported in a third party context or vice-versa. Should we prefix cookie names somehow to avoid confusion? e.g. 'firstparty-headerdata' and 'thirdparty-headerdata'?